packages/boba/teleportation/src/service.ts

```diff
@@ -0,0 +1,332 @@
+      )
+    }
+
+    // check if all chains are supported
+    // if the chain is supported, then store the contract of the chain and the balance info
+    // to the state
+    this.state.supportedChains = []
+    this.state.depositTeleportations = []
+    for (const chain of this.options.selectedBobaChains) {
+      const chainId = chain.chainId
+      const isSupported = await this.state.Teleportation.supportedChains(
+        chainId
+      )
+      if (!isSupported) {
+        throw new Error(
+          `Chain ${chainId} is not supported by the contract ${this.state.Teleportation.address}`
+        )
+      } else {
+        this.state.supportedChains = [...this.state.supportedChains, chain]
+        const depositTeleportation = new Contract(
+          chain.teleportationAddress,
+          TeleportationJson.abi,
+          chain.provider
+        )
+        const totalDisbursements =
+          await this.state.Teleportation.totalDisbursements(chainId)
+        const totalDeposits = await depositTeleportation.totalDeposits(
+          this.options.chainId
+        )
+        this.state.depositTeleportations.push({
+          Teleportation: depositTeleportation,
+          chainId,
+          totalDisbursements,
+          totalDeposits,
+          height: chain.height,
+        })
+      }
+    }
+  }
+
+  protected async _start(): Promise<void> {
+    while (this.running) {
+      for (const depositTeleportation of this.state.depositTeleportations) {
+        // search BobaReceived events
+        const latestBlock =
+          await depositTeleportation.Teleportation.provider.getBlockNumber()
+        try {
+          const events = await this._watchTeleportation(
+            depositTeleportation,
+            latestBlock
+          )
+          await this._disbureTeleportation(
+            depositTeleportation,
+            events,
+            latestBlock
+          )
+        } catch (err) {
+          this.logger.error('Error while running teleportation', {
+            err,
+          })
+        }
+      }
+      await sleep(this.options.pollingInterval)
+    }
+  }
+
+  async _watchTeleportation(
+    depositTeleportation: DepositTeleportations,
+    latestBlock: number
+  ): Promise<any> {
+    let lastBlock: number
+    const chainId = depositTeleportation.chainId.toString()
+    try {
+      lastBlock = await this._getDepositInfo(chainId)
+    } catch (e) {
+      this.logger.warn(`No deposit info found in chainId - ${chainId}`)
+      lastBlock = depositTeleportation.height
+      // store the new deposit info
+      await this._putDepositInfo(chainId, lastBlock)
+    }
+    const events = await this._getEvents(
+      depositTeleportation.Teleportation,
+      this.state.Teleportation.filters.BobaReceived(),
+      lastBlock,
+      latestBlock
+    )
+    return events
+  }
+
+  async _disbureTeleportation(
```

Comment:
```suggestion
  async _disburseTeleportation(
```
also, does this have a test?

Reply:
Fixed